patches/v8/0011-Use-target-cfg.patch

```diff
@@ -0,0 +1,423 @@
+             "//conditions:default": ":noicu/torque",
+         }),
+     )
+-    _v8_torque_definitions(
++    _v8_torque_outputs(
+         name = "icu/" + name,
+         prefix = "icu",
+         srcs = icu_srcs,
+         args = args,
+-        extras = extras,
++        definition_extras = definition_extras,
++        initializer_extras = initializer_extras,
+         tool = select({
+             "@v8//bazel/config:v8_target_is_32_bits": ":icu/torque_non_pointer_compression",
+             "//conditions:default": ":icu/torque",
+@@ -573,16 +517,19 @@ _v8_mksnapshot = rule(
+             mandatory = True,
+             allow_files = True,
+             executable = True,
+-            cfg = "exec",
++            cfg = get_cfg(),
+         ),
+         "target_os": attr.string(mandatory = True),
+-        "_allowlist_function_transition": attr.label(
+-            default = "@bazel_tools//tools/allowlists/function_transition_allowlist",
+-        ),
++        # "_allowlist_function_transition": attr.label(
++        #     default = "@bazel_tools//tools/allowlists/function_transition_allowlist",
++        # ),
+         "prefix": attr.string(mandatory = True),
+         "suffix": attr.string(mandatory = True),
+     },
+-    cfg = v8_target_cpu_transition,
++    # This allows specifying a CPU architecture on the command line to be used when compiling
++    # mksnapshot. Disable this functionality as we do not use it and it makes cfg changes more
++    # difficult.
++    # cfg = v8_target_cpu_transition,
+ )
+ 
+ def v8_mksnapshot(name, args, suffix = ""):
+@@ -703,3 +650,34 @@ def v8_build_config(name):
+         outs = ["icu/" + name + ".json"],
+         cmd = "echo '" + build_config_content(cpu, "true") + "' > \"$@\"",
+     )
++
++# Clone of genrule, but set up to compile for target configuration. Use with care, this may not
++# support all features of genrule(), but is sufficient for this use case.
++# Derived from the tensorflow project, see workerd build/run_binary_target.bzl for details.
++def _genrule_target_impl(ctx):
++    tool = ctx.attr.tool[DefaultInfo].files_to_run.executable
++    flags = [ ctx.expand_location(a) if "$(location" in a else a for a in ctx.attr.args ]
++
++    cmd = " ".join([tool.path] + flags)
++    ctx.actions.run_shell(
++        inputs = ctx.files.srcs,
++        outputs = ctx.outputs.outs,
++        tools = [tool],
++        use_default_shell_env = True,
++        command = cmd,
++    )
++
++genrule_target = rule(
++    implementation = _genrule_target_impl,
++    output_to_genfiles = True,
++    attrs = {
++        "outs": attr.output_list(mandatory = True),
++        "srcs": attr.label_list(allow_files = True),
++        "args": attr.string_list(),
++        "tool": attr.label(
++            executable = True,
++            cfg = get_cfg(),
++            mandatory = True
++        ),
++    },
++)
+-- 
+2.39.2 (Apple Git-143)
```

Comment:
JFYI, the v8-updates.md doc suggests using "git format-patch --full-index -k --no-signature".